sunpy/map/tests/test_map_factory.py

```diff
@@ -80,6 +80,10 @@ def test_patterns(self):
         eitmap = sunpy.map.Map(a_fname)
         assert isinstance(eitmap, sunpy.map.GenericMap)
 
+        # File name with file-type
+        eitmap = sunpy.map.Map(a_fname, filetype=pathlib.Path(a_fname).suffix[1:])
+        assert isinstance(eitmap, sunpy.map.GenericMap)
```

Comment:
You aren't really testing this properly here as you are giving it a filename which it can detect the type for. You would be better off giving it a random filename and the type.

(Although it will also use magic bytes to detect file type if the filename doesn't pan out so I am not sure exactly how to make this test fail.)